File: backend/src/main/kotlin/blue/mild/covid/vaxx/dao/model/InsuranceCompany.kt
package blue.mild.covid.vaxx.dao.model

enum class InsuranceCompany(
    val csFullName: String,
    val code: Int
) {
    VZP("Všeobecná zdravotní pojišťovna", 111),
    VOZP("Vojenská zdravotní pojišťovna", 201),
    CPZP("Česká průmyslová zdravotní pojišťovna", 205),
    OZP("Oborová zdravotní pojišťovna zaměstnanců bank, pojišťoven a stavebnictví", 207),
    ZPS("Zaměstnanecká pojišťovna Škoda", 209),
    ZPMV("Zdravotní pojišťovna ministerstva vnitra", 211),
    RBP("Revírní bratrská pokladna, zdravotní pojišťovna", 213)
}

File: frontend/src/app/model/InsuranceCompany.ts
export interface InsuranceCompany {
  code: string;
  numericCode?: number;
  name: string;
  csFullName: string;
}

export const insuranceNumericCodeMap: { [key: string]: number; } = {
  VZP: 111,
  VOZP: 201,
  CPZP: 205,
  OZP: 207,
  ZPS: 209,
  ZPMV: 211,
  RBP: 213
};
